.github/workflows/github-ci.yml

```diff
@@ -0,0 +1,69 @@
+name: Java CI with Maven and Release Artifact
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  release:
+    types: [ released ]
+
+permissions:
+  contents: write
+  actions: read
+  checks: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    outputs:
+      version: ${{ steps.extract_version.outputs.version }}
+      artifactId: ${{ steps.extract_artifactId.outputs.artifactId }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: '23'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Build with Maven
+        run: mvn -B clean install
+
+      - name: Extract project version
+        id: extract_version
+        run: |
+          VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+
+      - name: Extract artifactID
+        id: extract_artifactId
+        run: |
+          ARTIFACT_ID=$(mvn help:evaluate -Dexpression=project.artifactId -q -DforceStdout)
+          echo "artifactId=$ARTIFACT_ID" >> $GITHUB_OUTPUT
+
+      - name: Upload JAR
+        uses: actions/upload-artifact@v4
+        with:
+          name: java-artifact
+          path: target/${{ steps.extract_artifactId.outputs.artifactId }}-${{ steps.extract_version.outputs.version }}.jar
+
+  release:
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.event_name == 'release'
+    steps:
+      - uses: actions/download-artifact@v4
+        with:
+          name: java-artifact
+          path: target
+
+      - name: Upload Release Asset
+        uses: softprops/action-gh-release@v2
+        with:
+          files: target/${{ needs.build.outputs.artifactId }}-${{ needs.build.outputs.version }}.jar
```

Comment:
Just to understand: are the names of those variables `needs.build.outputs.*` referencing the names defined above in lines 21/22?